operator/pkg/apis/operator/v1alpha1/type.go

```diff
@@ -89,31 +93,38 @@ type ImageRegistry struct {
 type KarmadaComponents struct {
 	// Etcd holds configuration for etcd.
 	// +required
+	// +kubebuilder:default:={etcd: {}}
 	Etcd *Etcd `json:"etcd,omitempty"`
 
 	// KarmadaAPIServer holds settings to kube-apiserver component. Currently, kube-apiserver
 	// is used as the apiserver of karmada. we had the same experience as k8s apiserver.
 	// +optional
+	// +kubebuilder:default:={imageRepository: "registry.k8s.io", imageTag: "v1.25.2", replicas: 1}
 	KarmadaAPIServer *KarmadaAPIServer `json:"karmadaAPIServer,omitempty"`
 
 	// KarmadaAggregratedAPIServer holds settings to karmada-aggregated-apiserver component of the karmada.
 	// +optional
+	// +kubebuilder:default:={imageRepository: "docker.io", imageTag: "v1.4.0", replicas: 1}
 	KarmadaAggregratedAPIServer *KarmadaAggregratedAPIServer `json:"karmadaAggregratedAPIServer,omitempty"`
 
 	// KubeControllerManager holds settings to kube-controller-manager component of the karmada.
 	// +optional
+	// +kubebuilder:default:={imageRepository: "registry.k8s.io", imageTag: "v1.25.2", replicas: 1}
 	KubeControllerManager *KubeControllerManager `json:"kubeControllerManager,omitempty"`
 
 	// KarmadaControllerManager holds settings to karmada-controller-manager component of the karmada.
 	// +optional
+	// +kubebuilder:default:={imageRepository: "docker.io", imageTag: "v1.4.0", replicas: 1}
```

Comment:
It seems inconvenient that we need to replace the default version with every new Karmada version released. Is there a better way to do it? How about using cmd options like `karmadactl init`？

Reply:
I think we seem not need to add the imageTag field to every component, for user, they only care about the karmada version. 

whether we support global fields for karmada version?  and we calculate the corresponding to k8s version. It,s convenient for upgrading for users and us.  they only need to update one field, like this:

```golang
type KarmadaSpec struct {
  KarmadaVersion *string `json:"karmadaVersion,omitempty"`
}
```

Reply:
oh.. It,s seem two questions. and I agress with your suggest use `option` to set default version for karmada. we can specify the default installtion version with a flag.